### PR TITLE
Add Application Launcher Cluster, Audio Output Cluster, Media Input Cluster, Target Navigator Cluster, TV Channel Cluster, Wake on LAN Cluster, Keypad Input Cluster

### PR DESCRIPTION
**Pull request by lazarkov**
#### Problem
We do not have application launcher cluster which is needed for Chip to launch apps on TV
We do not have audio output cluster which is needed for Chip to control audio output of TV
We do not have media input cluster which is needed for Chip to control input of TV
We do not have target navigator cluster which is needed for Chip to navigate the apps on TV
We do not have tv channel cluster which is needed for Chip to control channels of TV
We do not have wake on LAN cluster which is needed for Chip to wake up TV from power sleep
We do not have keypad input cluster which is needed for Chip to make keypad commands on TV
We do not have account login cluster which is needed for Chip to login users to content app or node

#### Summary of Changes
_This is boilerplate code. There will be follow up with logic update of the cluster._
- Added application launcher server.cpp, audio output server.cpp, media input server.cpp, target navigator server.cpp, tv channel server.cpp, wake on LAN server.cpp, keypad input server.cpp, account login.cpp file
- Added application launcher cluster.xml, audio output cluster.xml, media input cluster.xml, target navigator cluster.xml, tv channel cluster.xml, wake on LAN cluster.xml, keypad input cluster.xml, account login .xml file    
- Updated example chip-tool so you can use all listed clusters
- Updated example tv-app so you can use it a TV server for use all listed clusters
- Updated all-cluster-app  so you can use it a server for use all listed clusters

#### Documentation
- You can see the application launcher cluster specification in this [URL](https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/1846)
- You can see the audio output cluster specification in this [URL](https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/275)
- You can see the media input cluster specification in this [URL](https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/274)
- You can see the target navigator cluster specification in this [URL](https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/1847)
- You can see the tv channel cluster specification in this [URL](https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/270)
- You can see the wake on LAN cluster specification in this [URL](https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/293)
- You can see the keypad input cluster specification in this [URL](https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/277)
- You can see the account login cluster specification in this [URL](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/c3d552880ed6fbb24bf4d85c94f8f842cd7875b7/src/app_clusters/media/AccountLogin.adoc)

#### Test
- Tested locally using chip-tool client and newly created tv-server
- Used the `./scripts/tools/zap_regen_all.py ` and `./gn_build.sh ` to verify the building is successful

### CODE REVIEW

**Comment by andy31415** [comment]
@lazarkov - could you mention the spec numbers where the clusters are defined? I was trying to follow the cluster definitions to check them against the spec but could not easily find them (and am not sure if I have an out of date version or not).

File: src/app/zap-templates/zcl/application-launcher-cluster.xml
Comment (andy31415) [inline src/app/zap-templates/zcl/application-launcher-cluster.xml:28]: should this be named 'catalog list' to match the spec?

I am unclear from the spec type `[list-def][uint16]` what type this should be. Could you explain this a bit?

File: src/app/zap-templates/zcl/application-launcher-cluster.xml
Comment (andy31415) [inline src/app/zap-templates/zcl/application-launcher-cluster.xml:46]: catalogVendorId seems to be uint16 in the spec

File: src/app/zap-templates/zcl/audio-output-cluster.xml
Comment (andy31415) [inline src/app/zap-templates/zcl/audio-output-cluster.xml:27]: Similar question here: how do we deal with list-def? I assume these are lists of a specific type rather than OCTET_STRING. Is this true? If this needs changes, could we add a GH issue to track required changes to support lists?

File: src/app/zap-templates/zcl/keypad-input-cluster.xml
Comment (andy31415) [inline src/app/zap-templates/zcl/keypad-input-cluster.xml:30]: I believe action/code are reversed here

File: src/app/zap-templates/zcl/target-navigator-cluster.xml
Comment (andy31415) [inline src/app/zap-templates/zcl/target-navigator-cluster.xml:37]: I cannot find the response/status definition in the latest build of the spec. Status makes sense to me, unsure what 'data' is supposed to contain.

Are we planning to update the spec or should we wait for the spec before adding this item?

File: src/app/zap-templates/zcl/tv-channel-cluster.xml
Comment (andy31415) [inline src/app/zap-templates/zcl/tv-channel-cluster.xml:28]: is the non-list attribute also a limitation of current zap (I.e. I have to use octet_string instead of actual LineupInfo/ChannelInfo type)?

File: src/app/zap-templates/zcl/tv-channel-cluster.xml
Comment (andy31415) [inline src/app/zap-templates/zcl/tv-channel-cluster.xml:64]: should we keep naming as in spec (which uses 'LinupInfo' instead of 'Lineup')?

File: src/app/zap-templates/zcl/wake-on-lan-cluster.xml
Comment (andy31415) [inline src/app/zap-templates/zcl/wake-on-lan-cluster.xml:27]: attribute in spec seems to be string rather than binary string. Is that supported for attributes?

**Review by andy31415 — COMMENTED** [review]
I compared XML cluster definitions vs a newly built master spec for `make html-all`.

I believe that is the best I can do  - json/zap seems to be periodically reorganized and generated files are too many to review content.

**Comment by andy31415** [comment]
I was also told some deltas will appear in the TV clusters: https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/2055/files

unsure how to handle them. Likely this PR does not need those changes as they are not merged in the spec yet, however we will have to do additional passes to later to refresh our definitions.

File: src/app/zap-templates/zcl/application-launcher-cluster.xml
Comment (lazarkov) [reply src/app/zap-templates/zcl/application-launcher-cluster.xml:28]: As far as I know, list is currently unsupported, this is "temporary" type until list is added.

File: src/app/zap-templates/zcl/audio-output-cluster.xml
Comment (lazarkov) [reply src/app/zap-templates/zcl/audio-output-cluster.xml:27]: If i'm not mistake this [commit](https://github.com/project-chip/connectedhomeip/pull/5020/commits/f68f3402f0263ed9b1024b63d8e6bc54aa9cb3c5) should add list to the CHIP.

File: src/app/zap-templates/zcl/keypad-input-cluster.xml
Comment (lazarkov) [reply src/app/zap-templates/zcl/keypad-input-cluster.xml]: Updated.

File: src/app/zap-templates/zcl/tv-channel-cluster.xml
Comment (lazarkov) [reply src/app/zap-templates/zcl/tv-channel-cluster.xml]: Updated.

File: src/app/zap-templates/zcl/wake-on-lan-cluster.xml
Comment (lazarkov) [reply src/app/zap-templates/zcl/wake-on-lan-cluster.xml]: Updated to CHAR_STRING.

File: src/app/zap-templates/zcl/target-navigator-cluster.xml
Comment (lazarkov) [reply src/app/zap-templates/zcl/target-navigator-cluster.xml:37]: Hi @chrisdecenzo can you provide clarification on data field of TargetNavigator Cluster?

**Comment by lazarkov** [comment]
> unsure how to handle them. Likely this PR does not need those changes as they are not merged in the spec yet, however we will have to do additional passes to later to refresh our definitions.

Yes, there will be update to the clusters after the spec PR is merged.

File: src/app/zap-templates/zcl/application-launcher-cluster.xml
Comment (lazarkov) [reply src/app/zap-templates/zcl/application-launcher-cluster.xml]: Updated.

File: src/app/zap-templates/zcl/tv-channel-cluster.xml
Comment (lazarkov) [reply src/app/zap-templates/zcl/tv-channel-cluster.xml:28]: I've fwd the question to the ZAP folks, I'm pretty sure it's unsupported today. 
Waiting for their confirmation.

**Review by andy31415 — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**
